test/EndpointTest.cpp

```diff
@@ -241,7 +241,7 @@ TEST_F(EndpointTest, get_hostnames)
         hostlist << host << "\n";
     }
     hostlist.close();
-    strncpy(data->hostlist_path, hostlist_path.c_str(), GEOPM_ENDPOINT_HOSTLIST_PATH_MAX);
+    strncpy(data->hostlist_path, hostlist_path.c_str(), GEOPM_ENDPOINT_HOSTLIST_PATH_MAX - 1);
     strncpy(data->agent, "monitor", GEOPM_ENDPOINT_AGENT_NAME_MAX);
```

Comment:
here too?  there are a bunch more if you grep for strncpy

Reply:
I just fixed the ones that gcc8 was complaining about.